packages/react-aria-components/src/Collection.tsx

```diff
@@ -484,6 +487,7 @@ export class Document<T, C extends BaseCollection<T>> extends BaseNode<T> {
   private subscriptions: Set<() => void> = new Set();
 
   constructor(collection: C) {
+    // @ts-ignore
```

Comment:
this one stumps me, why is this ok?

Reply:
It is a circular dependency... the `ownerDocument` needs to be itself, but `this` cannot be accessed until after calling `super`. It gets assigned in the initializer above, and the super constructor doesn't actually use it other than assigning so figured ignoring was ok.